packages/amplify_auth_cognito/android/src/main/kotlin/com/amazonaws/amplify/amplify_auth_cognito/types/FormatUserAttribute.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_auth_cognito.types
+
+import androidx.annotation.NonNull
+import com.amplifyframework.auth.AuthUserAttribute
+import com.amplifyframework.auth.AuthUserAttributeKey
+import java.lang.Error
+import java.lang.reflect.Method
+
+fun createAuthUserAttribute(key: String, value: String): AuthUserAttribute {
+    val authUserAttributeKey: AuthUserAttributeKey = createAuthUserAttributeKey(key);
+    return AuthUserAttribute(authUserAttributeKey, value)
+}
+
+fun createAuthUserAttributeKey(keyName: String): AuthUserAttributeKey {
+    when (keyName) {
```

Comment:
*Note:* Previously we were using reflection to convert the string into an `AuthUserAttributeKey`. I changed it to an explicit map for a few reasons.
1. Consistency with iOS. iOS previously wasn't using an associated value, which was problematic, so I updated it to be a map.
2. Differences between platforms. The two platforms have some differences that were not being accounted for. For example, iOS uses "birthDate" whereas android uses "birthdate". An explicit map allows us to account for those differences.
3. Personal preference. I think being explicit here is a lot easier to understand.

Reply:
I am assuming you retested the signUp flow, correct?